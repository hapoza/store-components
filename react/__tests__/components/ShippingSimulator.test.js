import React from 'react'
import { MockedProvider } from 'react-apollo/test-utils'
import { render } from 'intl-helper'

import ShippingSimulator from '../../ShippingSimulator'

describe('<ShippingSimulator /> component', () => {
  const renderComponent = (customProps = {}) => {
    const props = {
      country: 'brazil',
      ...customProps,
    }
    const component = (
      <MockedProvider mocks={[]}>
        <ShippingSimulator {...props} />
      </MockedProvider>
    )

    return render(component)
  }

  it('should be able to mount and not break', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toBeTruthy()
  })

  it('should match snapshot without skuId and seller', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with skuId and seller', () => {
    const props = {
      skuId: 'skuId',
      seller: 1,
    }
    const { asFragment } = renderComponent(props)
    expect(asFragment()).toMatchSnapshot()
  })
})
