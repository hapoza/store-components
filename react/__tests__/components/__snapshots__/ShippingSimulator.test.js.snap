// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShippingSimulator /> component should match snapshot with skuId and seller 1`] = `
<DocumentFragment>
  <form
    class="shippingContainer t-small c-on-base"
  >
    <label
      class="shippingZipcodeLabel c-muted-2 db t-small mb3"
      for="shipping-zipcode"
    >
      Calculate shipping
    </label>
    <div
      class="flex"
    >
      <input
        id="shipping-zipcode"
        name="zipcode"
        type="text"
        value=""
      />
      <button
        class="shippingCTA"
        disabled=""
        type="submit"
      >
        Ok
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`<ShippingSimulator /> component should match snapshot without skuId and seller 1`] = `
<DocumentFragment>
  <div
    class="shippingContainer"
  >
    <div
      class="content-loader-mock"
    >
      <rect
        height="100%"
        width="7em"
      />
      <rect
        height="100%"
        width="15em"
        x="8em"
      />
    </div>
  </div>
</DocumentFragment>
`;
